src/main/java/macrobase/analysis/pipeline/OneShotPipeline.java

```diff
@@ -0,0 +1,36 @@
+package macrobase.analysis.pipeline;
+
+import macrobase.analysis.result.AnalysisResult;
+import macrobase.conf.ConfigurationException;
+import macrobase.conf.MacroBaseConf;
+
+import java.io.IOException;
+import java.sql.SQLException;
+
+
+abstract public class OneShotPipeline extends AbstractPipeline {
```

Comment:
I'm suggesting having a wrapper around the entire pipeline do this for us. The main difference from the current approach is that the wrapper would keep track of the resultset last produced so that it could compute a diff, return the entire new result set, etc.